rasa/nlu/extractors/extractor.py

```diff
@@ -208,6 +210,18 @@ def convert_predictions_into_entities(
                 entities[-1][ENTITY_ATTRIBUTE_END] = token.end
                 if confidences is not None:
                     self._update_confidence_values(entities, confidences, idx)
+
+            elif self._check_is_single_entity(
```

Comment:
The check before, e.g. `elif token.start - last_token_end <= 1:`. should be covered by this as well.

Reply:
`_check_is_single_entity(...)` contains this check: `tokens_within_range = token.start - last_token_end <= 3` - is that what you meant?

Reply:
yes, but then we should remove the if part in line 209

Reply:
hmm, not sure I understand correctly - the former `if` statement catches things like compounding `New York`, and the new code addresses a slightly different case. Or do I misunderstand something here?

Reply:
That is correct, but your check should also cover that, right? And `_check_is_single_entity` should in my opinion cover all scenarios that indicates that the two tokens belong to one entity.  So I would just do something like
```
if new_tag_found:
    # new entity found, e.g. create new entity
    ...
elif self._check_is_single_entity(...):
    # same entity tags and same entity
    ...
else:
    # same entity tags but different entities
    ...
```

Reply:
ah, right, got it :) - thank you!